cocotb/regression.py

```diff
@@ -641,7 +641,9 @@ class TestFactory:
     """
 
     def __init__(self, test_function, *args, **kwargs):
-        if not isinstance(test_function, cocotb.coroutine):
+        if inspect.isasyncgenfunction(test_function):
+            raise TypeError(f"Found yield in async function '{test_function.__name__}'")
```

Comment:
We still support python 3.5, so you can't use f-strings yet.

Reply:
Ok, thanks the E999 error message from flake8 was not helpful. Nevertheless, it is not the issues I have mentioned. Let me fix this first.